samples/client/petstore/php/OpenAPIClient-php/lib/ObjectSerializer.php

```diff
@@ -346,6 +347,8 @@ public static function deserialize($data, $class, $httpHeaders = null)
         }
 
         if ($class === '\SplFileObject') {
+            $data = Psr7\Utils::streamFor($data);
```

Comment:
> Looks totally ok. Now I'm trying to imagine the use case... Hmm, deserialization, right? Then it should be response from remote server with already initiated `\StreamInterface` in response body. Maybe Guzzle can transcode response on the fly that way, but I don't know. Seems like it will never happen 🤣

I am going at this from the opposite end, if a user wants to pass an array of data to have the SDK instantiate the full object (including nested objects) for them.

API responses wouldn't ever contain an `SplFileObject` object, but a user can call

```php
require_once __DIR__ . "/vendor/autoload.php";

$config = OpenAPI\Client\Configuration::getDefaultConfiguration();
$config->setUsername("YOUR_API_KEY");

$api = new OpenAPI\Client\Api\FooRequestApi(
    new GuzzleHttp\Client(),
    $config
);

$data = [
    "file" => "/path/to/file.pdf",
    // ... other data here
];

$obj = ObjectSerializer::deserialize($data, ModelClass::class)

try {
    $result = $api->fooRequestSend($data);
    print_r($result);
} catch (Exception $e) {
    echo "Exception when calling API: " . $e->getMessage() . PHP_EOL;
}

```

I believe this is a preferable alternative to having to do:

```php
$obj = new ModelClass();
$obj->setFile(new SplFileObject("/path/to/file.pdf"))
    ->setProp1(/* ... */)
    ->setProp2(/* ... */)
    ->setProp3(/* ... */);
```

If a user has prebuilt JSON data they can simply pass the data to `ObjectSerializer::deserialize()` and everything _works_. Likewise with unit tests, now we can just have a bundle of pre-defined JSON data fixtures we can use to run many unit tests using data providers.